src/DotNet.Status.Web/Controllers/GitHubHookController.cs

```diff
@@ -28,14 +29,19 @@ public class GitHubHookController : ControllerBase
         private readonly IOptions<GitHubConnectionOptions> _githubOptions;
         private readonly ILogger<GitHubHookController> _logger;
         private readonly IGitHubApplicationClientFactory _gitHubApplicationClientFactory;
+        private readonly IAzureDevOpsClientFactory _azureDevOpsClientFactory;
+        private readonly IOptions<AzureDevOpsOptions> _azureDevOpsOptions;
         private readonly ITimelineIssueTriage _timelineIssueTriage;
         private readonly ITeamMentionForwarder _teamMentionForwarder;
+        private readonly Lazy<IAzureDevOpsClient> _azureClientLazy;
```

Comment:
I'm not sure if this will work with secret rotation... maybe it will?  But I don't think "IOptions" (or Lazy members) can react to new secret values.

Reply:
This is how we do it in the AzurePipelinesController, but... yeah, I don't know. I don't have to do it this way, probably.

Reply:
Moved away from the lazy thing to create the client at work item creation time.

Reply:
@alexperovich is my go to expert. This might explain some of our outages if the other things do this too. :-)

Reply:
anything that uses a secret, and isn't an exe or something, needs to access the secrets when it is used. This is an aspnet controller though, so you can just inject `IOptionsSnapshot<>` and use it. The controller lifetime is a single request.